File: frontend/src/app/services/child/child.service.ts
import { Injectable, inject } from '@angular/core';
import { UseQuery } from '@ngneat/query';
import { HttpClient } from '@angular/common/http';

const API_ENDPOINT = "http://localhost:8080/api" ;

interface IChildData {
  paidAmount: number;
  createdAt: string;
  parent: any;
}

@Injectable({
  providedIn: 'root'
})
export class ChildService {

  private useQuery = inject(UseQuery);
  private http = inject(HttpClient);

  constructor() { }

  // Retrieves child data for a given ID
  getChildData(parentId: number) {
    let page_num: number = 0;

    const queryKey = ['getChildData', page_num];

    const query =  this.useQuery(queryKey, () => {
      return this.http.post<IChildData[]>(
        `${API_ENDPOINT}/children/${page_num}`,
        {
          parentId
        }
      );
    }) as any;

    const invalidateQuery = (pageNum: number) => {
      page_num = pageNum;
      query.refetch();
    }
    
    return { query, invalidateQuery }
  }
}

File: backend/src/main/java/org/backend/service/ChildService.java
package org.backend.service;

import java.util.List;
import org.backend.model.Child;

public interface ChildService {
    // Retrieves a list of all children
    List<Child> getChildren();
    
    // Retrieves a list of children based on the parent ID
    // parent_id: the ID of the parent
    List<Child> getChildrenByParentId(Long parent_id);
}
